pkg/authorization/reverse_permission_claims_authorizer.go

```diff
@@ -0,0 +1,197 @@
+}
+
+func (a *ReversePermissionClaimsAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+	lcluster, err := genericapirequest.ClusterNameFrom(ctx)
+	if err != nil {
+		return authorizer.DecisionNoOpinion, "", err
+	}
+
+	apiBindings, err := a.getAPIBindingsReferenceForAttributes(attr, lcluster)
+	if err != nil {
+		return authorizer.DecisionNoOpinion, "", err
+	}
+
+	if len(apiBindings) == 0 {
+		return authorizer.DecisionAllow, "no API binding bound", nil
+	}
+
+	var authorizers []authorizer.Authorizer
+	for _, binding := range apiBindings {
+		for _, claim := range binding.Spec.PermissionClaims {
+			claim := claim
+			if claim.Group != attr.GetAPIGroup() || claim.Resource != attr.GetResource() {
+				continue
+			}
+			authorizers = append(authorizers, NewRestrictToVerbsAuthorizer(binding, &claim))
+		}
+	}
+
+	return IntersectionAuthorizer(authorizers).Authorize(ctx, attr)
+}
+
+type RestrictToVerbsAuthorizer struct {
+	binding *apisv1alpha1.APIBinding
+	claim   *apisv1alpha1.AcceptablePermissionClaim
+}
+
+func NewRestrictToVerbsAuthorizer(binding *apisv1alpha1.APIBinding, claim *apisv1alpha1.AcceptablePermissionClaim) *RestrictToVerbsAuthorizer {
+	return &RestrictToVerbsAuthorizer{
+		binding: binding,
+		claim:   claim,
+	}
+}
+
+func (a *RestrictToVerbsAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
+	if a.claim.Group != attr.GetAPIGroup() || a.claim.Resource != attr.GetResource() {
+		return authorizer.DecisionNoOpinion, fmt.Sprintf("unclaimed resource in API binding name=%q", a.binding.GetName()), nil
+	}
+
+	// claim is not accepted by the user that is importing it,
+	// hence the authorizer allows accessing the requested resource.
+	if a.claim.State != apisv1alpha1.ClaimAccepted {
+		return authorizer.DecisionAllow, fmt.Sprintf("claim is not accepted in API binding name=%q", a.binding.GetName()), nil
+	}
+
+	requestingAllResources := attr.GetName() == "" && attr.GetNamespace() == ""
+	resourceSelectorMatches := a.claim.All || requestingAllResources
+	for _, claimedResource := range a.claim.ResourceSelector {
+		var ns string
+		if claimedResource.Namespace != "" {
+			ns = attr.GetNamespace()
+		}
+		var name string
+		if claimedResource.Name != "" {
+			name = attr.GetName()
+		}
+		if claimedResource.Name == name && claimedResource.Namespace == ns {
+			resourceSelectorMatches = true
+			break
+		}
+	}
+
+	if !resourceSelectorMatches {
+		return authorizer.DecisionAllow, fmt.Sprintf("resource doesn't match any resource selector in API binding name=%q", a.binding.GetName()), nil
+	}
+
+	if len(a.claim.Verbs.RestrictTo) == 1 && a.claim.Verbs.RestrictTo[0] == "*" {
+		return authorizer.DecisionAllow, fmt.Sprintf("permission claim restricts to \"*\" in API binding name=%q", a.binding.GetName()), nil
```

Comment:
style nit: use back-tick quotes for the string to not need to escape `""` quotes, makes it slightly  easier to read